source/geometry_model/two_merged_boxes.cc

```diff
@@ -0,0 +1,523 @@
+                             "Number of cells in X direction in the lithosphere.");
+          prm.declare_entry ("Y repetitions lithosphere", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in Y direction in the lithosphere.");
+          prm.declare_entry ("Z repetitions lithosphere", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in Z direction in the lithosphere.");
+
+          // Whole box periodicity
+          prm.declare_entry ("X periodic", "false",
+                             Patterns::Bool (),
+                             "Whether the box should be periodic in X direction.");
+          prm.declare_entry ("Y periodic", "false",
+                             Patterns::Bool (),
+                             "Whether the box should be periodic in Y direction.");
+          prm.declare_entry ("Z periodic", "false",
+                             Patterns::Bool (),
+                             "Whether the box should be periodic in Z direction.");
+
+          prm.declare_entry ("X periodic lithosphere", "false",
+                             Patterns::Bool (),
+                             "Whether the box should be periodic in X direction in the lithosphere,");
+          prm.declare_entry ("Y periodic lithosphere", "false",
+                             Patterns::Bool (),
+                             "Whether the box should be periodic in Y direction in the lithosphere.");
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    TwoMergedBoxes<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Box with lithosphere boundary indicators");
+        {
+          const double thickness_lith = prm.get_double("Lithospheric thickness");
+
+          extents[0]           = prm.get_double ("X extent");
+          lower_extents[0]     = extents[0];
+          upper_extents[0]     = extents[0];
+          lower_box_origin[0]  = prm.get_double ("Box origin X coordinate");
+          upper_box_origin[0]  = lower_box_origin[0];
+          periodic[0]          = prm.get_bool ("X periodic");
+          periodic[dim]        = prm.get_bool ("X periodic lithosphere");
+          lower_repetitions[0] = prm.get_integer ("X repetitions");
+          upper_repetitions[0] = prm.get_integer ("X repetitions lithosphere");
+          // Note that when the number of repetitions in the horizontal direction(s)
+          // between the upper and lower box differ, an internal boundary remains
+          // between the two boxes. Hence the assert here and later.
+          AssertThrow(lower_repetitions[0] == upper_repetitions[0],
+                      ExcMessage("Mantle and lithosphere must have the same number of repetitions in the X-direction."));
```

Comment:
Ah, and here is the corresponding assertion. 

If the two have to be the same, why not just call it `X repetitions` and use it for both parts of the domain?